org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateAdaptorServletListener.mtl

```diff
@@ -140,17 +152,26 @@ public class [javaClassNameForAdaptorServletListener(anAdaptorInterface) /] impl
     }
 
     private static String generateBasePath(final ServletContextEvent servletContextEvent) {
-        final ServletContext servletContext = servletContextEvent.getServletContext();
-        String base = getInitParameterOrDefault(servletContext, PROPERTY_BASE, DEFAULT_BASE);
-        return base + servletContext.getContextPath();
+        return (getBasePathFromEnvironment().orElseGet(() -> getBasePathFromContext(servletContextEvent).orElseGet(() -> DEFAULT_BASE))) +
```

Comment:
It would be better to combine them with UriBuilder to avoid double slashes.

Reply:
True. shall fix,